### PR TITLE
Delayed toggle indication

### PR DESCRIPTION
**Pull request by stevenocchipinti**
I have the following situation:
- A model has a boolean field set to `true`
- A toggle is rendered based on that model (with `checked = true`)
- The toggle is clicked which causes an update request be sent to the server
- While there has not yet been a response, the user doesn't get any visual feedback
- When a response is received, the model is updated
- The toggle is re-rendered with `checked = false`

In the first commit of this PR, I have added an example to demonstrate this.

As the user is left without any feedback for their action, in the second commit of this PR I have made a modification that works something like this:
- A toggle is rendered with `checked = true`
- When the toggle is clicked, internal state is updated (to `false`)
- If the internal state !== the `checked` prop
  - Set a 'pending' CSS class
- If the internal state === the `checked` prop
  - Remove the 'pending' CSS class

This allows the stylesheet to dictate what the switch looks like after the user has toggled it and before the checked prop has been updated.
I have provided CSS rules that uses a pulsing effect while it is pending, but other options could be replacing the tick or cross with a small spinner or setting a color while pending, etc.

This functionality is controlled by the `allowPending` option to maintain a backward compatible behavior, specifically with the _Controlled Component without onChange_ scenario, which would normally cause a read-only toggle.

Hopefully my explanation and examples make sense, I have a little more detail in the commit messages.
I would be interested to know what you think!

-- Steve

### CODE REVIEW

File: index.es6.js [hunk 1]
```diff
@@ -51,7 +52,7 @@ export default React.createClass({
       return
     }
 
-    if (!('checked' in this.props)) {
+    if (!('checked' in this.props) || this.state.pendingAllowed) {
```
Comment (ahstro) [inline index.es6.js:55]: This should probably use `this.props.allowPending` directly, as I'm not seeing any other uses of `this.state.pendingAllowed`

File: index.es6.js [hunk 2]
```diff
@@ -31,6 +31,7 @@ export default React.createClass({
     }
     return {
       checked: !!checked,
+      pendingAllowed: !!this.props.allowPending,
```
Comment (ahstro) [inline index.es6.js:34]: Should probably be removed?

**Review by ahstro — CHANGES_REQUESTED** [review]
This is a neat feature and I'd like to see it merged. I didn't have much time, so I haven't played around with it and tested it thoroughly, but if you rebase it to resolve the merge conflicts, I'll take a look soon :)

And thank you for the wonderful commit messages!

File: index.es6.js (hunk 1 above)
Comment (stevenocchipinti) [reply index.es6.js:55]: I agree, state is not needed here

**Comment by stevenocchipinti** [comment]
You're welcome :)

I've opened PR #69 to replace this one because the code has moved on quite a bit since this was opened.

**Pull request closed without merging**